ring-jetty-adapter/src/ring/adapter/jetty.clj

```diff
@@ -28,11 +28,37 @@
           (servlet/update-servlet-response response response-map)
           (.setHandled base-request true))))))
 
-(defn- ^AbstractHandler async-proxy-handler [handler timeout]
+(defn- attach-async-timeout-listener
+  [^HttpServletRequest request
```

Comment:
We could write this:

```clojure
(defn- async-timeout-listener [request context response handler]
```

And then type hint just when using `.sendError` and `complete`.